src/server.js

```diff
@@ -151,13 +151,13 @@ module.exports = function (classes) {
               self._checkAuth(req, socket).then(function (result) {
                 if (result) {
                   return self.handleWebsocket(req, socket, body).then(function(result){
-                    return conn.handleMessage(result);
+                    return self.conn.handleMessage(result);
```

Comment:
`conn` here is a local variable, it's not a member from the `Endpoint` prototype.  i was trying to think a better way to do this (like pass the `{conn, result}`) to the handleMessage callback, but it's an chicken/egg problem (and a sign of spaghetti code) from my code